### PR TITLE
Add watches for files or directories

### PR DESCRIPTION
**Pull request by blag**
This allows users to add normal files to be observed, instead of only directories, when using the inotify observer.

### CODE REVIEW

**Comment by BoboTiG** [comment]
Nice!
Could you add the feature everywhere and tests please?

**Comment by blag** [comment]
@BoboTiG Done. I also included a fix I should have added to the `python2.7-support` branch.

**Comment by BoboTiG** [comment]
Thanks, I will review today.
For the Python 2.7 fixes, I already pushed a fix yesterday. Could you rebase your branch on master?

FYI you can launch tests more easily like:

    python2 setup.py test
    python3 setup.py test

**Comment by blag** [comment]
@BoboTiG Rebased on `master` and removed my duplicate fixes - should be good to go (as long as tests pass).

File: tests/test_emitter.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 from watchdog.events import *
 from watchdog.observers.api import ObservedWatch
 from watchdog.utils import platform
-from watchdog.utils.compat import Queue
+from watchdog.utils.compat import queue
```
Comment (BoboTiG) [inline tests/test_emitter.py]: Can you delete this change?

File: tests/test_emitter.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def setup_function(function):
     global p, event_queue
     tmpdir = os.path.realpath(mkdtemp())
     p = partial(os.path.join, tmpdir)
-    event_queue = Queue()
+    event_queue = queue.Queue()
```
Comment (BoboTiG) [inline tests/test_emitter.py]: Idem.

File: tests/test_inotify_c.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 from watchdog.events import DirCreatedEvent, DirDeletedEvent, DirModifiedEvent
 from watchdog.observers.api import ObservedWatch
 from watchdog.utils import platform
-from watchdog.utils.compat import Queue
+from watchdog.utils.compat import queue
```
Comment (BoboTiG) [inline tests/test_inotify_c.py]: Idem.

File: tests/test_inotify_c.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ def setup_function(function):
     global p, event_queue
     tmpdir = os.path.realpath(mkdtemp())
     p = partial(os.path.join, tmpdir)
-    event_queue = Queue()
```
Comment (BoboTiG) [inline tests/test_inotify_c.py]: Idem.

**Comment by blag** [comment]
Fixed.

**Comment by BoboTiG** [comment]
Can you rebase from master? We will benefit of Travis-CI this time :muscle: 
BTW I think you contributed enough to be listed as contributor, if you want to ;)

**Comment by BoboTiG** [comment]
Is it possible to have the feature on Windows too?

File: tests/test_emitter.py [hunk 5]
```diff
@@ -35,7 +35,13 @@
     from watchdog.observers.inotify import InotifyFullEmitter
 elif platform.is_darwin():
     pytestmark = pytest.mark.skip('WATCHDOG-8')
```
Comment (BoboTiG) [inline tests/test_emitter.py]: Because of this line, tests are skipped on macOS :see_no_evil: 
Could you have working tests when you remove it?

Comment (blag) [reply tests/test_emitter.py]: > Because of this line, tests are skipped on macOS 🙈

Yeah, but Travis was all green, so the PR was good to go! 🤣

Comment (blag) [reply tests/test_emitter.py]: I'm working on adding tests for all of the backends I have access to. Will hopefully have them soon.

**Comment by BoboTiG** [comment]
Just a little note to let you know we are archiving this repository. The original repo is being alive now, we should not maintain a fork then.

Thanks, and perhaps your patch would be a good fit for the original repos.

**Comment by BoboTiG** [comment]
Closing as archived project.

**Pull request closed without merging**